src/client/mod.rs

```diff
@@ -213,7 +213,13 @@ impl Client {
         }
 
         Err(ErrorKind::ServerSelectionError {
-            message: "timed out while trying to select server".into(),
+            message: self
```

Comment:
Rather than just putting in a long string here, what are your thoughts about putting a copy of the hashmap of server descriptions as a second field here and then using that as part of the display implementation on the ErrorKind variant? This might make it easier for users if they want to do some sort of processing on the error.

Reply:
We decided offline that users likely won't need programmatic access to the topology, but that more information should be included in the error message. To that end, I added all of the non-nil fields of a `ServerInfo` as well as compatibility error messages, if present, to the message.